kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -143,28 +133,13 @@ public State next() {
     }
   }
 
-  private void updateRotationTimers() {
-    long lastRotate = time.milliseconds();
-    if (log.isDebugEnabled() && rotateIntervalMs > 0) {
-      log.debug("Update last rotation timer. Next rotation for {} will be in {}ms", tp, rotateIntervalMs);
-    }
-    if (rotateScheduleIntervalMs > 0) {
-      nextScheduledRotate = DateTimeUtils.getNextTimeAdjustedByDay(lastRotate, rotateScheduleIntervalMs, timezone);
```

Comment:
Why wouldn't `rotate.interval.ms` default to `-1`, which will disable using wallclock time rotation entirely? Maybe part of the confusion here is that in the HDFS connector we only had the concept of wallclock time so anything time-based is wallclock time but the S3 connector uses the timestamp from the record? In which case the docs for `rotate.interval.ms` taken from HDFS don't make sense since the same implementation cannot accomplish the stated goal of `rotate.interval.ms`.